src/sources/kubernetes_logs/pod_metadata_annotator.rs

```diff
@@ -72,6 +75,9 @@ impl PodMetadataAnnotator {
 
         annotate_from_file_info(log, &self.fields_spec, &file_info);
         annotate_from_metadata(log, &self.fields_spec, &pod.metadata);
+        if let Some(ref pod_status) = pod.status {
```

Comment:
Nit: to match the `Pod` fields definition, let's move this block to the very end of this fn, below the `pod.spec` processing.
Fields to in the order of `metadata` -> `spec` -> `status`.